### PR TITLE
Openvino ep nuget

### PR DESCRIPTION
**Pull request by sfatimar**
These nuget changes introduces openvino libs in openvino flavor of ORT Nuget Package

**Motivation and Context**
- This change was requested by Adobe customers who did not want to set openvino libs seperately  when building and running csharp samples 
- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

**Comment by oliviajain** [comment]
@sfatimar: a few style errors causing python checks to fail

```
[flake8 PEP8 ERROR] ./tools/nuget/generate_nuspec_for_native_nuget.py:468:120: W291 trailing whitespace
[flake8 PEP8 ERROR] ./tools/nuget/generate_nuspec_for_native_nuget.py:469:38: E128 continuation line under-indented for visual indent
[flake8 PEP8 ERROR] ./tools/nuget/generate_nuspec_for_native_nuget.py:475:13: E265 block comment should start with '# '
[flake8 PEP8 ERROR] ./tools/nuget/generate_nuspec_for_native_nuget.py:477:30: E128 continuation line under-indented for visual indent
[flake8 PEP8 ERROR] ./tools/nuget/generate_nuspec_for_native_nuget.py:478:13: E265 block comment should start with '# '
```

**Comment by sfatimar** [comment]
> 

File updated

File: tools/nuget/generate_nuspec_for_native_nuget.py [hunk 1]
```diff
@@ -468,7 +493,7 @@ def generate_files(list, args):
                           nuget_dependencies['cuda_ep_shared_lib']) +
                           runtimes_target + args.target_architecture + '\\native" />')
 
-    # process all other library dependencies
+        # process all other library dependencies
```
Comment (jywu-msft) [inline tools/nuget/generate_nuspec_for_native_nuget.py:496]: spurious spaces/formatting?

File: tools/ci_build/build.py [hunk 2]
```diff
@@ -1822,8 +1822,12 @@ def build_nuget_package(source_dir, build_dir, configs, use_cuda, use_openvino,
             run_subprocess(cmd_args, cwd=csharp_build_dir)
 
         if is_windows():
-            # this path is setup by cmake/nuget_helpers.cmake for MSVC on Windows
-            nuget_exe = os.path.normpath(os.path.join(native_dir, config, "nuget_exe", "src", "nuget.exe"))
+            if use_openvino:
```
Comment (jywu-msft) [inline tools/ci_build/build.py:1825]: provide more details on why this if block is needed for openvino?
why doesn't the else path work?

Comment (sfatimar) [reply tools/ci_build/build.py:1825]: The offline option to build nuget packages --build_nuget does not build a directory nuget_exe/src/nuget.exe... I suppose the directory is created through some other ci pipelines to build nuget packages on window. We have not explored to the build pipelines for now . Here is an explanation from Scott where he mentions why a directory is needed
"In our build pipeline, where this script is primarily used, we assembly native libraries from various platforms (linux, mac, mobile) in the nuget-artifacts directory and package those. I don’t know the history as to why we implicitly determine it’s that sort of build by the presence of the nuget-artifacts directory. That could probably be made more explicit and/or use a different directory for the artifacts dir to what the local build does so they don’t clash."

File: tools/nuget/generate_nuspec_for_native_nuget.py (hunk 1 above)
Comment (sfatimar) [reply tools/nuget/generate_nuspec_for_native_nuget.py]: Spurious space removed

File: tools/ci_build/build.py (hunk 2 above)
Comment (jywu-msft) [reply tools/ci_build/build.py:1825]: thanks for the explanation!

**Review by jywu-msft — APPROVED** [review]

**Comment by MaajidKhan** [comment]
@oliviajain Hi olivia. can we add the release tag 1.10 for this PR if it's already not too late.
This has the latest c# nuget fix.

**Comment by oliviajain** [comment]
> @oliviajain Hi olivia. can we add the release tag 1.10 for this PR if it's already not too late. This has the latest c# nuget fix.

Hi Maajid, the 1.10 release branch has been finalized and is not accepting any more changes.

**Pull request merged**